From 34a871b9e2c6ceb84a92de5cafacf8e9d3974281 Mon Sep 17 00:00:00 2001
From: Antonio Borneo <antonio.borneo@foss.st.com>
Date: Thu, 6 Feb 2025 10:00:55 +0100
Subject: [PATCH 1641/1695] plat-stm32mp2: remove the driver
 stm32mp2x-pwr-irq-user

It is replaced by the generic driver stm32-irq-notifier

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git f09429f21f090f6297ab927d89f6191538a2e5f4 plat-stm32mp2: remove the driver stm32mp2x-pwr-irq-user]
Signed-off-by: Antonio Borneo <antonio.borneo@foss.st.com>
Change-Id: I08c8a073ab914213f877ea5ad6ee1c5e792e4455
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/440872
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
---
 .../plat-stm32mp2/drivers/stm32mp25_pwr_irq.c | 66 -------------------
 1 file changed, 66 deletions(-)

diff --git a/core/arch/arm/plat-stm32mp2/drivers/stm32mp25_pwr_irq.c b/core/arch/arm/plat-stm32mp2/drivers/stm32mp25_pwr_irq.c
index cf5ff6219..9837bca78 100644
--- a/core/arch/arm/plat-stm32mp2/drivers/stm32mp25_pwr_irq.c
+++ b/core/arch/arm/plat-stm32mp2/drivers/stm32mp25_pwr_irq.c
@@ -431,69 +431,3 @@ TEE_Result stm32mp25_pwr_irq_probe(const void *fdt, int node)
 
 	return TEE_SUCCESS;
 }
-
-static enum itr_return pwr_it_user_handler(struct itr_handler *handler)
-{
-	uint32_t *it_id = handler->data;
-
-	VERBOSE_PWR("pwr irq tester handler");
-
-	if (!it_id)
-		return ITRR_NONE;
-
-	notif_send_it(*it_id);
-
-	return ITRR_HANDLED;
-}
-
-static TEE_Result stm32mp25_pwr_irq_user_dt_probe(const void *fdt, int node,
-						  const void *compat __unused)
-{
-	uint32_t *optee_it_notif_id = NULL;
-	struct itr_chip *itr_chip = NULL;
-	TEE_Result res = TEE_SUCCESS;
-	const fdt32_t *cuint = NULL;
-	size_t itr_num = 0;
-
-	res = interrupt_dt_get(fdt, node, &itr_chip, &itr_num);
-	if (res)
-		return res;
-
-	cuint = fdt_getprop(fdt, node, "st,notif-it-id", NULL);
-	if (cuint) {
-		optee_it_notif_id = calloc(1, sizeof(*optee_it_notif_id));
-		if (!optee_it_notif_id)
-			return TEE_ERROR_OUT_OF_MEMORY;
-
-		*optee_it_notif_id = fdt32_to_cpu(*cuint);
-	}
-
-	res = interrupt_create_handler(itr_chip, itr_num, pwr_it_user_handler,
-				       optee_it_notif_id, 0, NULL);
-	if (res) {
-		free(optee_it_notif_id);
-		return res;
-	}
-
-	interrupt_enable(itr_chip, itr_num);
-	if (fdt_getprop(fdt, node, "wakeup-source", NULL)) {
-		if (interrupt_can_set_wake(itr_chip))
-			interrupt_set_wake(itr_chip, itr_num, true);
-		else
-			DMSG("PWR-IRQ wakeup source ignored");
-	}
-
-	return TEE_SUCCESS;
-}
-
-static const struct dt_device_match pwr_irq_test_match_table[] = {
-	{ .compatible = "st,stm32mp21-pwr-irq-user" },
-	{ .compatible = "st,stm32mp25-pwr-irq-user" },
-	{ }
-};
-
-DEFINE_DT_DRIVER(stm32mp25_pwr_irq_dt_tester) = {
-	.name = "stm32mp25-pwr-irq-user",
-	.match_table = pwr_irq_test_match_table,
-	.probe = &stm32mp25_pwr_irq_user_dt_probe,
-};
-- 
2.47.2

